### PR TITLE
Fix RTMP buffer ingest issues

### PR DESCRIPTION
**Pull request by SelimEmre**

### CODE REVIEW

File: src/main/java/io/antmedia/muxer/MuxAdaptor.java [hunk 1]
```diff
@@ -641,11 +641,10 @@ else if (bufferTimeMs > 0)
 							buffering = true;
 							//if buffer duration somehow is more than 5 times bufferTimeMs
 							logger.warn("Buffer is increased({}) too much for stream: {}", getBufferedDurationMs() ,streamId);
-							AVPacket pkt;
 							int i = 0;
-							while ((pkt = bufferQueue.poll()) != null) 
+							while ((bufferQueue.poll()) != null) 
 							{
-								pkt.close();
+								bufferQueue.poll().close();
```
Comment (mekya) [inline src/main/java/io/antmedia/muxer/MuxAdaptor.java:647]: Each `poll()` get the header of the queue. It means you get two packets at each time. 
One in `while ((bufferQueue.poll()) != null) ` and one  is  `bufferQueue.poll().close();`

File: src/main/java/io/antmedia/muxer/MuxAdaptor.java [hunk 2]
```diff
@@ -1099,7 +1098,10 @@ public void writeBufferedPacket()
 						if (pktTimeDifferenceMs < passedTime) 
 						{
 							writePacket(inputFormatContext.streams(tempPacket.stream_index()), tempPacket);
-							av_packet_unref(tempPacket);
+							if(tempPacket != null && tempPacket.size() >= 0) {
```
Comment (mekya) [inline src/main/java/io/antmedia/muxer/MuxAdaptor.java:1101]: I think this is just a patch solution. Do we have to use it? 

If possible, find out the correct cause for this problem. 

Secondly, maybe latest RTMP changes fix this problem. I'm not sure. You may test it.

**Review by mekya — CHANGES_REQUESTED** [review]

File: src/main/java/io/antmedia/muxer/MuxAdaptor.java (hunk 2 above)
Comment (SelimEmre) [reply src/main/java/io/antmedia/muxer/MuxAdaptor.java:1101]: I tested this issue in our latest SNAPSHOT. It seems it's solved with our latest changes.

File: src/main/java/io/antmedia/muxer/MuxAdaptor.java (hunk 1 above)
Comment (mekya) [reply src/main/java/io/antmedia/muxer/MuxAdaptor.java:647]: There is still problem here

**Review by mekya — CHANGES_REQUESTED** [review]

**Comment by mekya** [comment]
Closing this issue because of this comment 
https://github.com/ant-media/Ant-Media-Server/issues/2540#issuecomment-733197782

**Pull request closed without merging**